### ISSUE TITLE
[MARTIFACT-38] Show failure reason and fail build if a plugin is marked as "fail", even if no solution available

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-artifact-plugin#137]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** opened **[MARTIFACT-38](https://issues.apache.org/jira/browse/MARTIFACT-38?redirect=false)** and commented

currently, plugin shows a warning and continues:

```
[WARN] plugin without solution org.springframework.boot:spring-boot-maven-plugin:2.5.5
```

it would be better to report ERROR and the corresponding plugin issue (given it's provided in plugin issues config file: https://github.com/apache/maven-artifact-plugin/blob/master/src/main/resources/org/apache/maven/plugins/artifact/buildinfo/not-reproducible-plugins.properties )

```
[ERROR] plugin without solution org.springframework.boot:spring-boot-maven-plugin:2.5.5, see https://github.com/spring-projects/spring-boot/issues/21005
```

---

**Affects:** 3.3.0

**Remote Links:**
- [GitHub Pull Request #10
](https://github.com/apache/maven-artifact-plugin/pull/10)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Output is now:

```
osipovmi@deblndw011x:~/var/Projekte/lda-cafe/backend-services/service-controller ((v2.4) %)
$ mvn org.apache.maven.plugins:maven-artifact-plugin:3.3.1-SNAPSHOT:check-buildplan
[INFO] Scanning for projects...
[INFO]
[INFO] --------------< com.siemens.lda.cafe:service-controller >---------------
[INFO] Building service-controller 2.4
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-artifact-plugin:3.3.1-SNAPSHOT:check-buildplan (default-cli) @ service-controller ---
[INFO] no known issue with org.apache.maven.plugins:maven-enforcer-plugin:3.1.0
[INFO] no known issue with org.jacoco:jacoco-maven-plugin:0.8.3
[INFO] no known issue with org.apache.maven.plugins:maven-resources-plugin:3.3.0
[INFO] no known issue with org.apache.maven.plugins:maven-compiler-plugin:3.10.1
[INFO] no known issue with org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M7
[INFO] no known issue with org.apache.maven.plugins:maven-jar-plugin:3.3.0 (>= 3.2.0)
[ERROR] plugin without solution org.springframework.boot:spring-boot-maven-plugin:2.5.5, see https://github.com/spring-projects/spring-boot/issues/21005
[INFO] no known issue with org.apache.maven.plugins:maven-install-plugin:3.1.0
[INFO] no known issue with org.apache.maven.plugins:maven-deploy-plugin:3.0.0
[INFO] current module pom.xml is /var/osipovmi/Projekte/lda-cafe/backend-services/service-controller/pom.xml
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.354 s
[INFO] Finished at: 2023-02-01T15:27:28+01:00
[INFO] ------------------------------------------------------------------------
```

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

I fully agree on adding a link to the plugin issue: let's show where the root cause is and help people contribute on the issue

on failing the build, I understand the objective of showing clearly to users not reading warnings that the build is not reproducible

but I have 2 problems:
1. in case of multi-module build, the detection of issues is stopped by failing on the first module that has an issue: this does not permit to have a global view
2. for some plugins, there are workarounds that you can apply (see post-process cases, often with maven-replacer-plugin in https://github.com/jvm-repo-rebuild/reproducible-central/blob/master/doc/BUILDSPEC.md ): failing will block these users

the minimum I think is to add a property to disable fail on non-reproducible plugins

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with  [acc0cdb44f9e395c3cec995b270af7e728a70a1d](https://gitbox.apache.org/repos/asf?p=maven-artifact-plugin.git;a=commit;h=acc0cdb44f9e395c3cec995b270af7e728a70a1d).
